lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -42,33 +43,39 @@ import Cardano.Wallet.Primitive.AddressDerivation
     ( Depth (..), DerivationIndex, Passphrase )
 import Cardano.Wallet.Primitive.CoinSelection.Balance
     ( SelectionLimit, SelectionResult, SelectionSkeleton )
+import Cardano.Wallet.Primitive.Slotting
+    ( PastHorizonException )
 import Cardano.Wallet.Primitive.Types
-    ( PoolId, ProtocolParameters, SlotNo (..), TokenBundleMaxSize (..) )
+    ( PoolId
+    , ProtocolParameters
+    , SlotNo (..)
+    , TokenBundleMaxSize (..)
+    , WalletId
+    )
 import Cardano.Wallet.Primitive.Types.Address
     ( Address (..) )
 import Cardano.Wallet.Primitive.Types.Coin
     ( Coin (..) )
 import Cardano.Wallet.Primitive.Types.RewardAccount
     ( RewardAccount )
 import Cardano.Wallet.Primitive.Types.Tx
-    ( SealedTx (..)
-    , SerialisedTx (..)
-    , TokenBundleSizeAssessor
+    ( TokenBundleSizeAssessor
     , Tx (..)
     , TxConstraints
+    , TxIn
     , TxMetadata
     , TxOut
     )
-import Data.ByteString
-    ( ByteString )
 import Data.List.NonEmpty
     ( NonEmpty )
 import Data.Text
     ( Text )
+import Fmt
+    ( Buildable (..), genericF )
 import GHC.Generics
     ( Generic )
 
-data TransactionLayer k = TransactionLayer
+data TransactionLayer k tx = TransactionLayer
```

Comment:
I am intrigued about the direction of this `tx` parameter, but it seems you are not really leveraging it in this pr, right? There is no `mapTx :: (tx1 -> tx2) -> TransactionLayer k tx1 -> TransactionLayer k tx2` or similar, and you only use it with `SealedTx`?

Reply:
Yes, you are right - this type parameter isn't used by this PR. I am hoping to be able to:
1. Swap the SealedTx type for a mock type in unit testing.
2. Remove the dependency on cardano-api in cardano-wallet-core.